src/sim_engine/filesystem.rs

```diff
@@ -25,4 +25,11 @@ impl Filesystem for SimFilesystem {
         };
         Box::new(filesystem_copy)
     }
+    fn get_id(&self) -> String {
+        self.mount_point.clone()
+    }
+
+    fn eq(&self, other: &Filesystem) -> bool {
+        self.get_id() == other.get_id()
```

Comment:
When you clone a String you make a copy. This one is about efficiency...it seems wrong to make two exact copies and compare them when you can just compare the two things directly.